mlflow/diviner.py

```diff
@@ -0,0 +1,452 @@
+             wrapped libraries (i.e., ``GroupedProphet.predict_groups()``).
+          *  If the ``DataFrame`` supplied does not contain the column name ``groups``, then the
+             specific forecasting method that is primitive-driven (for ``GroupedProphet``, the
+             ``predict()`` method mirrors that of ``Prophet``'s, requiring a ``DataFrame``
+             submitted with explicit datetime values per group which is not a tenable
+             implementation for pyfunc or RESTful serving) is utilized. For ``GroupedProphet``,
+             this is the ``.forecast()`` method, while for ``GroupedPmdarima``, this is the
+             ``.predict()`` method.
+
+        :param dataframe: A ``pandas.DataFrame`` that contains the required configuration for the
+                          appropriate ``Diviner`` type.
+
+                          For example, for ``GroupedProphet.forecast()``:
+
+                          - horizon : int
+                          - frequency: str
+
+                          predict_conf = pd.DataFrame({"horizon": 30, "frequency": "D"}, index=[0])
+                          forecast = pyfunc.load_pyfunc(model_uri=model_path).predict(predict_conf)
+
+                          Will generate 30 days of forecasted values for each group that the model
+                          was trained on.
+        :return: A Pandas DataFrame containing the forecasted values for each group key that was
+                 either trained or declared as a subset with a ``groups`` entry in the ``dataframe``
+                 configuration argument.
+        """
+        schema = dataframe.columns.values.tolist()
+
+        conf = dataframe.to_dict(orient="index").get(0)
+
+        # required parameter extraction and validation
+        horizon = conf.get("horizon", None)
+        n_periods = conf.get("n_periods", None)
+        if n_periods and horizon and n_periods != horizon:
+            raise MlflowException(
+                "The provided prediction configuration contains both `n_periods` and `horizon` "
+                "with different values. Please provide only one of these integer values."
+            )
+        else:
+            if not n_periods and horizon:
+                n_periods = horizon
+
+        if not n_periods:
+            raise MlflowException(
+                "The provided prediction configuration Pandas DataFrame does not contain either "
+                "the `n_periods` or `horizon` columns. At least one of these must be specified "
+                f"with a valid integer value. Configuration schema: {schema}."
+            )
+
+        if not isinstance(n_periods, int):
+            raise MlflowException(
+                "The `n_periods` column contains invalid data. Supplied type must be an integer. "
+                f"Type supplied: {type(n_periods)}"
+            )
+
+        frequency = conf.get("frequency", None)
+
+        if self.diviner_instance_type == "GroupedProphet" and not frequency:
```

Comment:
Can we use `isinstance` here?

Reply:
Yep! I had removed the logic that I was using with the instance name and forgot to convert this back to something more standard. Changed!